### ISSUE TITLE
bug: Webdriver Jasmine expect(element).toBePresent() doesnt work with latest Appium 2.0 beta48 build

### ISSUE THREAD

**Issue by gforepsly** [appium/appium#18036]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Writing close like this inside IT block doesn't work (the test is running normally, but when it's time to check for presence, it fails on that step, saying the element is not present - the usual fail):
`expect($('~Workday history')).toBePresent()`

Error Example (a real mess):
`Error: Expected Element({ sessionId: '5ca4d36b-e54d-41aa-a6bd-d32742c7a490', elementId: '0C000000-0000-0000-CC1E-000000000000', element-6066-11e4-a52e-4f735466cecf: '0C000000-0000-0000-CC1E-000000000000', selector: '~Get started', parent: Browser({ sessionId: '5ca4d36b-e54d-41aa-a6bd-d32742c7a490', capabilities: Object({ webStorageEnabled: false, locationContextEnabled: false, browserName: '', platform: 'MAC', javascriptEnabled: true, databaseEnabled: false, takesScreenshot: true, networkConnectionEnabled: false, platformName: 'iOS', iosInstallPause: 5000, deviceName: 'iPhone 14 Pro', language: 'en', platformVersion: '16.2', automationName: 'XCUITest', app: '/Users/gfor/Documents/AutomationProject/TestAutomation/appium/apps/Repsly.app', newCommandTimeout: 10000, wdaLaunchTimeout: 300000, wdaConnectionTimeout: 600000, wdaStartupRetries: 4, wdaStartupRetryInterval: 120000, wdaLocalPort: 8668, udid: '96E22F2B-5DA1-4C51-973F-08E38416AA22' }), addCommand: Function, overwriteCommand: Function, addLocato ... to be present.`

Writing close like this inside IT block works (no fails, all good):
`expect($('~Workday history').isExisting).toBe(true)`

Error log when this fails (good logs):
`Error: Element isn't visible after 15000 seconds. Error: Error: element ("~Allow Once") still not displayed after 15000ms`

We are using webdriverIO in sync mode, it's a typescript project, appium server 2.0.0-beta.48 under MAC 13.1 Ventura. I will provide more configuration stuff below. 

Can anyone tell me what are we doing wrong? Why are this check above failing, but passing when we change the syntax a bit? We are still on older Node and older sync packages.

### Expected Behavior

`expect($('~Workday history')).toBePresent()` 
The above line should work, alongside with the other methods for toBeDisplayed etc. 

### Minimal Reproducible Example

tsconfig file

`{
  "compilerOptions": {
    "outDir": "./.tsbuild/",
    "target": "es2016",
    "module": "commonjs",
    "types": ["node", "webdriverio/sync", "@wdio/jasmine-framework"],
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "./appium/tests/**/*.ts"
  ]
}`

package.json

`"devDependencies": {
    "@babel/cli": "^7.20.7",
    "@babel/core": "^7.20.12",
    "@babel/eslint-parser": "^7.19.1",
    "@babel/preset-env": "^7.20.2",
    "@babel/register": "^7.18.9",
    "@babel/traverse": "^7.20.12",
    "@babel/types": "^7.20.7",
    "@ffmpeg-installer/ffmpeg": "^1.1.0",
    "@wdio/cli": "7.29.1",
    "@wdio/jasmine-framework": "7.26",
    "@wdio/local-runner": "7.29.1",
    "@wdio/spec-reporter": "7.29.1",
    "@wdio/types": "7.26.0",
    "@wdio/sync": "7.27",
    "axios": "0.27.2",
    "dotenv": "^16.0.1",
    "downloads-folder": "^3.0.3",
    "eslint": "^8.31.0",
    "eslint-config-standard": "^17.0.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^6.1.1",
    "eslint-plugin-wdio": "^7.25.3",
    "exceljs": "^4.3.0",
    "faker": "5.5.3",
    "file-type": "^16.0.1",
    "form-data": "^4.0.0",
    "jsdom": "^19.0.0",
    "moment": "^2.29.3",
    "node-fetch": "^2.6.1",
    "nodemailer": "^6.7.5",
    "pkce-challenge": "^3.0.0",
    "request": "^2.88.2",
    "testcafe": "^2.1.0",
    "testcafe-reporter-allure-expanded": "^1.1.0",
    "testcafe-reporter-teamcity": "^1.0.12",
    "testcafe-reporter-xunit": "^2.2.1",
    "ts-node": "^10.9.1",
    "typescript": "^4.9.4",
    "uuid": "^8.3.2",
    "wdio-teamcity-reporter": "^3.1.0",
    "webdriverio": "7.29.1",
    "xlsx": "^0.18.5"
  }`

babel.config.js

`module.exports = {
    presets: [
        [ '@babel/preset-env', {
            targets: {
                node: 'current',
            },
        } ],
    ]
};
`

### Environment

- Operating system: MAC M1 Pro on Ventura 13.1 OSX
- If running via `appium` CLI...
  - Appium CLI version (output of `appium --version`): 2.0.0-beta.48
  - Node.js version (output of `node --version`): v14.21.2
  - `npm` version (output of `npm --version`):  9.2.0
  - Last `appium` version which did _not_ exhibit the problem: 1.22.3
- If running Appium Desktop, its version:
- Appium driver(s) and their version(s): xcuitest@4.16.6 (automationName 'XCUITest')
- Appium plugin(s) and their version(s): no plugins installed
- Platform and version under test: IOS 16.2
- Real device or emulator/simulator: simulator iphone 14 Pro

### Link to Appium Logs

[AppiumLogs.zip](https://github.com/appium/appium/files/10400930/AppiumLogs.zip)

### Futher Information

Test files and appium server logs attached. 
I can't provide ios.app file, but you can find "Repsly" under app store as an app and simply use that one (it's public) for purpose of reproducing this one.

**Comment by jlipps** [comment]
We don't handle support for webdriverio or jasmine here. If you think you might have encountered an Appium bug then please link your Appium log. Otherwise this seems like a webdriverio question.

**Comment by gforepsly** [comment]
> We don't handle support for webdriverio or jasmine here. If you think you might have encountered an Appium bug then please link your Appium log. Otherwise this seems like a webdriverio question.

Thx for the quick reply!
I cannot provide some company-related info, but I will try and get you an appium log tomorrow for this behaviour.

**Comment by gforepsly** [comment]
@jlipps @mykola-mokhnach Hi!
I managed to get the appium logs, so I stored them under the first post. Please kindly check them all please tell me if it's possible to see anything from them.
BUG01 is the issue I was talking about
BUG02 is how it works properly.

**Comment by jlipps** [comment]
I don't see any issues in the Appium logs. In the error case it looks like you have some non sequenced commands, like you call get screenshot and then before waiting for the response try to find some elements. This would be due to a coding  error on your side or a misunderstanding of how async commands work in your libraries somehow. Nothing I see here makes me think that Appium is involved in the problem you're experiencing.

Also, for what it's worth, I don't typically use displayedness checks with Appium unless you have a real reason to. Typically presence checks are enough unless you suspect the element might actually be present but not displayed. Just waiting for the element to be present might be sufficient for your case.

**Comment by gforepsly** [comment]
@jlipps Like mentioned, we are using WebdriverIO in SYNC mode, it auto-resolves promises. I can write the test in async mode as well, and I will get the same results (I tried). So, I don't think it's an issue connected to coding.

**Comment by jlipps** [comment]
well, it's right there in the log:

```
2023-01-12 11:01:47:543 [HTTP] --> GET /wd/hub/session/5ca4d36b-e54d-41aa-a6bd-d32742c7a490/element/0C000000-0000-0000-CC1E-000000000000/displayed
...
2023-01-12 11:01:47:594 [HTTP] <-- GET /wd/hub/session/5ca4d36b-e54d-41aa-a6bd-d32742c7a490/element/0C000000-0000-0000-CC1E-000000000000/displayed 200 50 ms - 65
2023-01-12 11:01:47:594 [HTTP] 
2023-01-12 11:01:47:607 [HTTP] --> GET /wd/hub/session/5ca4d36b-e54d-41aa-a6bd-d32742c7a490/screenshot
...
2023-01-12 11:01:47:609 [HTTP] --> POST /wd/hub/session/5ca4d36b-e54d-41aa-a6bd-d32742c7a490/elements
2023-01-12 11:01:47:609 [HTTP] {"using":"accessibility id","value":"Get started"}
...
2023-01-12 11:01:47:728 [HTTP] <-- GET /wd/hub/session/5ca4d36b-e54d-41aa-a6bd-d32742c7a490/screenshot 200 119 ms - 353552
...
2023-01-12 11:01:47:803 [HTTP] <-- POST /wd/hub/session/5ca4d36b-e54d-41aa-a6bd-d32742c7a490/elements 200 194 ms - 139
```

Your script calls `displayed`, gets the result, then calls `screenshot`. without waiting for a result it also calls `find element`. commands should not be interleaved as it could cause undefined behaviour, though in this case i don't see any problems.

in sum, appium is doing exactly what your code is asking it to do, and responding with correct information. unless you can help me see something else here it's pretty conclusive that appium is not at fault here. maybe there's an issue with webdriverio's sync mode or how you're using it. i never use webdriverio's sync mode for what it's worth.

**Comment by gforepsly** [comment]
@jlipps So, I did a lot of tries and I can say that the "sync" option in WebdriverIO is the cause of the issues. As soon as I installed everything from scratch and started fully with async (also preferred) WebdriverIO way - all of "expect" assertions are working as expected.

The problem is that our project has more than 2 years, so with every WebdriverIO major version update something new was different, which ultimately now means that even with Webdriver 7.x and NodeJs 14 (last versions where sync syntax should work) it doesn't work all they way. We are refactoring our code and that will solve all of our isses.

Thx for the tips - We also check for presence before some actions like click etc.; this example was just for the sake of example.
